### ISSUE TITLE
RMap.readAllKeySet and  RMap.keySet in kvrocks problem

### ISSUE THREAD

**Issue by darkice1** [redisson/redisson#4624]
When using RMap.readAllKeySet to get the key, I can get the data normally. But using RMap.keySet will not be able to get the normal key.
The content and key are serialized (org.redisson.codec.KryoCodec). Except for the keySet error, all other cases can be used normally.
The error message is as follows.

```
Caused by: java.lang.NumberFormatException: For input string: "0043d5903a01b9d9ab46d4b474532bc�"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.lang.Long.parseLong(Long.java:578)
        at java.lang.Long.valueOf(Long.java:803)
        at org.redisson.client.codec.LongCodec$1.decode(LongCodec.java:38)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:374)
        at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:418)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:383)
        at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:196)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:134)
        at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:104)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507)
        at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)2022-10-25 13:14:32 [redisson-nett
```
config配置
```
sentinelServersConfig:
  idleConnectionTimeout: 10000
  connectTimeout: 1000
  timeout: 600000
  retryAttempts: 3
  retryInterval: 1000
  password: "xxxxxxxx"
  subscriptionsPerConnection: 400
  sslEnableEndpointIdentification: true
  sslProvider: "JDK"
  pingConnectionInterval: 1000
  keepAlive: true
  tcpNoDelay: false
  loadBalancer: !<org.redisson.connection.balancer.RoundRobinLoadBalancer> {}
  slaveConnectionMinimumIdleSize: 5
  slaveConnectionPoolSize: 300
  failedSlaveReconnectionInterval: 3000
  failedSlaveCheckInterval: 10000
  masterConnectionMinimumIdleSize: 5
  masterConnectionPoolSize: 200
  readMode: "SLAVE"
  subscriptionMode: "SLAVE"
  subscriptionConnectionMinimumIdleSize: 1
  subscriptionConnectionPoolSize: 50
  dnsMonitoringInterval: 5000
  sentinelAddresses:
  - "redis://192.168.1.131:17662"
  - "redis://192.168.1.127:17662"
  - "redis://192.168.1.130:17662"
  natMapper: !<org.redisson.api.DefaultNatMapper> {}
  masterName: "master"
  database: 0
  scanInterval: 1000
  checkSentinelsList: true
threads: 16
nettyThreads: 64
codec: !<org.redisson.codec.KryoCodec> {}
referenceEnabled: true
transportMode: "NIO"
lockWatchdogTimeout: 10000
keepPubSubOrder: true
useScriptCache: false
minCleanUpDelay: 5
maxCleanUpDelay: 1800
cleanUpKeysAmount: 100
nettyHook: !<org.redisson.client.DefaultNettyHook> {}
useThreadClassLoader: true
addressResolverGroupFactory: !<org.redisson.connection.DnsAddressResolverGroupFactory> {}
```

**Comment by git-hulk** [comment]
@darkice1 After looking through the Redisson issues, I think you should use English to describe your problem.

I noticed this issue coz other guys pinged offline. Does this happen in Kvrocks instead of Redis? If yes, I guess it's caused by Kvrocks can will return the string iterator instead of the number like Redis.

**Comment by darkice1** [comment]
> @darkice1 After looking through the Redisson issues, I think you should use English to describe your problem.
> 
> I noticed this issue coz other guys pinged offline. Does this happen in Kvrocks instead of Redis? If yes, I guess it's caused by Kvrocks can will return the string iterator instead of the number like Redis.

Thank you very much . I did used kvrocks.   Is there a way for me to solve this?

**Comment by git-hulk** [comment]
> > @darkice1 After looking through the Redisson issues, I think you should use English to describe your problem.
> > I noticed this issue coz other guys pinged offline. Does this happen in Kvrocks instead of Redis? If yes, I guess it's caused by Kvrocks can will return the string iterator instead of the number like Redis.
> 
> Thank you very much . I did used kvrocks. Is there a way for me to solve this?

Hi @darkice1, I'm not familiar with Java and Redisson, so to see if any Redisson community folk can help to confirm this issue.

**Comment by darkice1** [comment]
> 

I get it . Thank you very much.

**Comment by git-hulk** [comment]
@mrniko Hello Nikita, not sure if it's right to ping you here. My question is whether `RMap.Keyset` will use `HSCAN` to iter the hash key-values? if yes, this issue is caused by [Kvrocks](https://github.com/apache/incubator-kvrocks) will return the string cursor when using scan.

**Comment by git-hulk** [comment]
@mrniko Many thanks for your nice confirmation, Kvrocks scan command did return the string instead of the number cursor. The scan command is the only one that Kvrocks don't compatible with Redis. Do you have any suggestions on this? @mrniko

**Comment by mrniko** [comment]
@git-hulk 

Fixed in https://github.com/redisson/redisson/issues/5517

**Comment by git-hulk** [comment]
> @git-hulk @darkice1
> 
> Fixed in #5517

@mrniko That's cool. Thank you!
